src/plugins/collision/collider/backend.rs

```diff
@@ -131,6 +131,74 @@ impl<C: ScalableCollider> Plugin for ColliderBackendPlugin<C> {
                 );
             });
 
+        // When the `Sensor` component is added to a collider,
+        // remove the collider's contribution on the rigid body's mass properties.
+        app.observe(
+            |trigger: Trigger<OnAdd, Sensor>,
+             query: Query<(
+                &ColliderParent,
+                &ColliderMassProperties,
+                &PreviousColliderTransform,
+            )>,
+             mut body_query: Query<MassPropertiesQuery>| {
+                if let Ok((
+                    collider_parent,
+                    collider_mass_properties,
+                    previous_collider_transform,
+                )) = query.get(trigger.entity())
+                {
+                    if let Ok(mut mass_properties) = body_query.get_mut(collider_parent.0) {
+                        // Subtract previous collider mass props from the body's own mass props.
+                        mass_properties -= ColliderMassProperties {
+                            center_of_mass: CenterOfMass(
+                                previous_collider_transform
+                                    .transform_point(collider_mass_properties.center_of_mass.0),
+                            ),
+                            ..*collider_mass_properties
+                        };
+                    }
+                }
+            },
+        );
+
+        // When the `Sensor` component is removed from a collider,
+        // add the collider's mass properties to the rigid body's mass properties.
+        app.observe(
+            |trigger: Trigger<OnRemove, Sensor>,
+             mut collider_query: Query<(
+                Ref<C>,
+                &ColliderParent,
+                &ColliderDensity,
+                &mut ColliderMassProperties,
+                &ColliderTransform,
+            )>,
+             mut body_query: Query<MassPropertiesQuery>| {
+                if let Ok((
+                    collider,
+                    collider_parent,
+                    density,
+                    mut collider_mass_properties,
+                    collider_transform,
+                )) = collider_query.get_mut(trigger.entity())
+                {
+                    if let Ok(mut mass_properties) = body_query.get_mut(collider_parent.0) {
+                        // Update collider mass props.
+                        *collider_mass_properties =
+                            collider.mass_properties(density.max(Scalar::EPSILON));
+
+                        // Add new collider mass props to the body's mass props.
+                        mass_properties += ColliderMassProperties {
```

Comment:
These lines are duplicated. I recommend adding the following function:
```rust
impl ColliderMassProperties {
  fn with_collider_transform(self, transform: &ColliderTransform) -> Self {
    Self {
      center_of_mass: CenterOfMass(
        collider_transform.transform_point(self.center_of_mass.0),
      ),
      ..self
    }
  }
}
```

Reply:
Good idea! I named it `transformed_by` though to be more similar to Bevy's transformation methods on e.g. `Mesh` and bounding volumes